ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivCall.java

```
@@ -0,0 +1,93 @@
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.privacy.methods.priv;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcErrorConverter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.AbstractBlockParameterMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.BlockParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.JsonCallParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.privacy.PrivateTransactionSimulator;
+import org.hyperledger.besu.ethereum.transaction.CallParameter;
+
+public class PrivCall extends AbstractBlockParameterMethod {
+
+  private final PrivateTransactionSimulator privateTransactionSimulator;
+
+  public PrivCall(
+      final BlockchainQueries blockchainQueries,
+      final PrivateTransactionSimulator privateTransactionSimulator) {
+    super(blockchainQueries);
+    this.privateTransactionSimulator = privateTransactionSimulator;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.PRIV_CALL.getMethodName();
+  }
+
+  @Override
+  protected BlockParameter blockParameter(final JsonRpcRequestContext request) {
+    return request.getRequiredParameter(2, BlockParameter.class);
```

We should probably make this an optional parameter (it defaults to "latest").

Also, while the reorg handling in private state, we can only support latest. Should we fail the request if the user specifies anything that isn't the head? My concern is that if we silently accept something else but return the result for the latest block things might look weird.

---

There is a check now that the latest block was specified for the priv_call. See method resultByBlockNumber in PrivCall.java